### PR TITLE
Add a test procedure in nyc_taxis workload for measuring performance for Searchable Snapshot feature

### PR DESCRIPTION
**Pull request by tlfeng**
### Description
Add a workload for measuring performance for Searchable Snapshot feature.
The new workload is based on "[nyc_taxis](https://github.com/opensearch-project/opensearch-benchmark-workloads/tree/main/nyc_taxis)" workload, after adding snapshot operations and removing unnecessary test procedures.

The code is an outcome of the plan I proposed in issue https://github.com/opensearch-project/OpenSearch/issues/4797#issuecomment-1301606437, aims to add first-class support to run performance test for Searchable Snapshot.

Related to issue https://github.com/opensearch-project/OpenSearch/issues/3983, improvement of benchmarking specific features is needed.

### Issues Resolved
https://github.com/opensearch-project/OpenSearch/issues/5218

### Testing
Ran the command to start a performance test:
`opensearch-benchmark execute_test --workload-path=~/github/opensearch-benchmark-workloads/searchable_snapshot --distribution-version=2.4.0 --provision-config-instance="1gheap" --provision-config-instance-params="~/workload/searchable-snapshot/provision_config_instance_params.json" --opensearch-plugins="repository-s3" --plugin-params="~/workload/searchable-snapshot/plugin_params.json" --workload-params="~/workload/searchable-snapshot/workload_params.json" --kill-running-processes --test-mode`

Result:
```
   ____                  _____                      __       ____                  __                         __
  / __ \____  ___  ____ / ___/___  ____ ___________/ /_     / __ )___  ____  _____/ /_  ____ ___  ____ ______/ /__
 / / / / __ \/ _ \/ __ \\__ \/ _ \/ __ `/ ___/ ___/ __ \   / __  / _ \/ __ \/ ___/ __ \/ __ `__ \/ __ `/ ___/ //_/
/ /_/ / /_/ /  __/ / / /__/ /  __/ /_/ / /  / /__/ / / /  / /_/ /  __/ / / / /__/ / / / / / / / / /_/ / /  / ,<
\____/ .___/\___/_/ /_/____/\___/\__,_/_/   \___/_/ /_/  /_____/\___/_/ /_/\___/_/ /_/_/ /_/ /_/\__,_/_/  /_/|_|
    /_/

[INFO] Preparing for test execution ...
[INFO] Executing test with workload [searchable_snapshot], test_procedure [searchable_snapshot] and provision_config_instance ['1gheap'] with version [2.4.0].

Running delete-index                                                           [100% done]
Running create-index                                                           [100% done]
Running check-cluster-health                                                   [100% done]
Running index                                                                  [100% done]
Running refresh-after-index                                                    [100% done]
Running force-merge                                                            [100% done]
Running refresh-after-force-merge                                              [100% done]
Running wait-until-merges-finish                                               [100% done]
Running create-snapshot-repository                                             [100% done]
Running create-snapshot                                                        [100% done]
Running wait-for-snapshot-create                                               [100% done]
Running delete-local-index                                                     [100% done]
Running restore-snapshot                                                       [100% done]
Running default                                                                [100% done]
Running range                                                                  [100% done]
Running distance_amount_agg                                                    [100% done]
Running autohisto_agg                                                          [100% done]
Running date_histogram_agg                                                     [100% done]
...
```

Note:
1. `refresh` operation is necessary after indexing data, otherwise there will be error "Failed to snapshot shard" during taking snapshot.
2. `force-merge` before taking snapshot aims to reduce the amount of segments in snapshot and increase the search performance.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by tlfeng** [comment]
@achitojha @IanHoang @travisbenedict I would like to hear your opinion as maintainers. Do you think it's reasonable to add a workload for testing performance of a specific feature, like the requested change here?

**Comment by tlfeng** [comment]
@andrross Could you please take a look at the procedures for the performance test and the Readme file, does it satisfy the demand?

**Comment by engechas** [comment]
Given that this workload utilizes many of the same operations and the same corpus as nyc_taxis, what do you think about adding this as a new test procedure to nyc_taxis rather than a new track? There are a number of existing test procedures in nyc_taxis for reference.

https://github.com/opensearch-project/opensearch-benchmark-workloads/blob/main/nyc_taxis/test_procedures/default.json#L2
https://github.com/opensearch-project/opensearch-benchmark-workloads/blob/main/nyc_taxis/test_procedures/default.json#L129
https://github.com/opensearch-project/opensearch-benchmark-workloads/blob/main/nyc_taxis/test_procedures/default.json#L185

**Comment by tlfeng** [comment]
@engechas Thank you for the review! 👍 I agree that add the steps for benchmarking Searchable Snapshots feature into a "test procedure" is an option, and it avoids duplicating the codes.
I will modify the PR to add a new test procedure in "nyc_taxis" workload. My concern for going that way is this is the procedure to test a specific feature, not a general benchmarking, which is different from existing workloads and adding new "test procedure" into may make them messy.

**Comment by tlfeng** [comment]
@engechas @travisbenedict Could you review my PR again when you have time? I have updated the PR to add a new "test procedure" in "nyc_taxis" workload for testing the performance of Searchable Snapshot feature, instead of a new workload. Thank you!

File: nyc_taxis/TEST_PROCEDURES.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Test Procedures
+
+## searchable_snapshot
```
Comment (engechas) [inline nyc_taxis/TEST_PROCEDURES.md:3]: Could you add documentation about the supported OpenSearch versions for this feature?

File: nyc_taxis/TEST_PROCEDURES.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# Test Procedures
+
+## searchable_snapshot 
+
+A test procedure for measuring performance for Searchable Snapshot feature. It runs the same search query as the default test procedure `append-no-conflicts`.
+
+In contrast with `append-no-conflicts` which running query on index with local storage, this test procedure runs query on index backed by a remote snapshot.
+
+The remote snapshot stores in Amazon S3, an Amazon S3 bucket for storing the snapshot and an AWS account user credential that has permission to access the bucket is required for the workload,
+to learn more in configuring Amazon S3 bucket as a snapshot repository, see the [OpenSearch docs](https://opensearch.org/docs/latest/opensearch/snapshots/snapshot-restore#amazon-s3).
```
Comment (engechas) [inline nyc_taxis/TEST_PROCEDURES.md:10]: Maybe reword to something like
```
This test procedure requires that a remote snapshot be stored in Amazon S3. To learn more about configuring Amazon S3 as a snapshot repository, see the [OpenSearch docs](https://opensearch.org/docs/latest/opensearch/snapshots/snapshot-restore#amazon-s3)
```

File: nyc_taxis/operations/default.json [hunk 3]
```diff
@@ -25,6 +25,47 @@
...
     },
+    {
+      "name": "create-snapshot-repository",
+      "operation-type": "create-snapshot-repository",
+      "repository": "{{ snapshot_repository_name | default('test-repository') }}",
+      "body": {
+        "type": "s3",
+        "settings": {
+          "bucket": "{{ s3_bucket_name | default('opensearch-snapshot') }}",
+          "region": "{{ s3_bucket_region | default('us-east-1') }}"
```
Comment (engechas) [inline nyc_taxis/operations/default.json:36]: Should we provide defaults here? The bucket needs to be pre-created AFAIK so if it is not and the `s3_bucket_name` parameter is omitted, then the track will fail once it reaches this operation. We should be able to fail faster by making these required parameters.

Same comment for any parameters that require user setup

File: nyc_taxis/operations/default.json [hunk 4]
```diff
@@ -25,6 +25,47 @@
...
+        }
+      }
+    },
+    {
+      "name": "create-snapshot",
+      "operation-type": "create-snapshot",
+      "repository": "{{ snapshot_repository_name | default('test-repository') }}",
+      "snapshot": "{{ snapshot_name | default('test-snapshot') }}",
+      "body": {
+        "indices": []
```
Comment (engechas) [inline nyc_taxis/operations/default.json:46]: Should the `nyc_taxis` index be in this array?

Same question for the `restore-snapshot` operation

File: nyc_taxis/TEST_PROCEDURES.md (hunk 1 above)
Comment (tlfeng) [reply nyc_taxis/TEST_PROCEDURES.md:3]: Thank you so much for your review! @engechas 👍👍 I will address the comments one by one.

Comment (tlfeng) [reply nyc_taxis/TEST_PROCEDURES.md:3]: Added the supported version in line 11 (https://github.com/opensearch-project/opensearch-benchmark-workloads/pull/58/files#diff-a3c7b0ad3c2082729bd2ee692decaa774510b3ffa044732d4e37d8445d1f2cc1R11). It's a good idea to explicitly mention the version since because the feature is not supported from the beginning.

File: nyc_taxis/TEST_PROCEDURES.md (hunk 2 above)
Comment (tlfeng) [reply nyc_taxis/TEST_PROCEDURES.md]: Thanks for the advice, I made a modification to the expression to make it clearer.
`The test procedure will create a remote snapshot that stored in Amazon S3, so an Amazon S3 bucket for storing the snapshot and an AWS account user credential that has permission to access the bucket are required to run the test procedure. To learn more about configuring Amazon S3 as a snapshot repository, see the [OpenSearch docs](https://opensearch.org/docs/latest/opensearch/snapshots/snapshot-restore#amazon-s3).`

File: nyc_taxis/operations/default.json (hunk 3 above)
Comment (tlfeng) [reply nyc_taxis/operations/default.json]: Thanks! This is a very useful idea! I removed the default value of `s3_bucket_name` and `s3_bucket_region`, and updated the documentation in commit https://github.com/opensearch-project/opensearch-benchmark-workloads/pull/58/commits/53178a45dfa7bdcd7ef66831d53bb962c22e7f94.

File: nyc_taxis/operations/default.json (hunk 4 above)
Comment (tlfeng) [reply nyc_taxis/operations/default.json:46]: According to the doc (https://www.elastic.co/guide/en/elasticsearch/reference/7.10/create-snapshot-api.html#create-snapshot-api-request-body):
> By default, a snapshot includes all data streams and indices in the cluster. If this argument is provided, the snapshot only includes the specified data streams and clusters.

There 2 reasons I didn't specify the index name in the `indices` parameter:
1. There will only be one index during the benchmark run, because there is an operation to delete all the index as the first step - https://github.com/opensearch-project/opensearch-benchmark-workloads/blob/1f85d5c56f586e1dcc787d8093055ba69f79451c/nyc_taxis/test_procedures/default.json#L7
I don't think it's necessary to add the index name.
2. I didn't see any workloads have got hard-coded index name in the test procedure definitions, and I didn't find a predefined variable name that stands for the index name of the workload, so I'd rather not hard-coding the index name here.

Comment (engechas) [reply nyc_taxis/operations/default.json:46]: > There will only be one index during the benchmark run, because there is an operation to delete all the index as the first step

I don't think this is true, and if it is then we need to fix that as it could inadvertently delete important data.

IIRC, OSB only deletes the test index. For example, if I run SO, then run NYC_TAXIS, the NYC_TAXIS run will not delete the index created for SO. Users may have data on the cluster outside of the test index, by not specifying the test index here, we are adding to the duration of the test (and potentially skewing metrics) as the snapshot and restore will occur for all indices rather than just the NYC_TAXIS test index.

Comment (tlfeng) [reply nyc_taxis/operations/default.json:46]: @engechas Sorry didn't see your comment in time. Thanks for your explanation, and I will add the index name explicitly.

Comment (tlfeng) [reply nyc_taxis/operations/default.json:46]: You are right, based on the document (https://esrally.readthedocs.io/en/stable/track.html#delete-index), `delete-index` operation (by default) will only delete the indices that specified in the workload.

**Comment by tlfeng** [comment]
@IanHoang @gkamat Could you share your opinions on my PR when you have time? I heard you are maintaining this repository. Thank you!

**Comment by gkamat** [comment]
@tlfeng, taking a look, will update with any comments shortly.

File: nyc_taxis/README.md [hunk 5]
```diff
@@ -74,6 +74,9 @@ This workload allows to overwrite the following parameters using `--workload-par
 * `target_throughput` (default: default values for each operation): Number of requests per second, `none` for no limit.
 * `search_clients`: Number of clients that issues search requests.
 
+### Test Procedures
+The workload contains different test procedures, see [TEST_PROCEDURES](TEST_PROCEDURES.md) for details.
```
Comment (gkamat) [inline nyc_taxis/README.md:78]: This workload contains multiple test procedures, ...

File: nyc_taxis/TEST_PROCEDURES.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
+# Test Procedures
+
+## searchable_snapshot 
+
+A test procedure for measuring performance for Searchable Snapshots feature. It runs the same search query as the default test procedure `append-no-conflicts`.
```
Comment (gkamat) [inline nyc_taxis/TEST_PROCEDURES.md:5]: for the Searchable Snapshots feature.  It runs the same search queries as ...

File: nyc_taxis/TEST_PROCEDURES.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+In contrast with `append-no-conflicts` which running query on index with local storage, this test procedure runs query on index backed by a remote snapshot.
```
Comment (gkamat) [inline nyc_taxis/TEST_PROCEDURES.md:7]: which runs queries on an  index stored on the cluster itself, this test procedure runs queries on an index ...

File: nyc_taxis/TEST_PROCEDURES.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+The test procedure will create a remote snapshot that stored in Amazon S3, so an Amazon S3 bucket for storing the snapshot and an AWS account user credential that has permission to access the bucket are required to run the test procedure. To learn more about configuring Amazon S3 as a snapshot repository, see the [OpenSearch docs](https://opensearch.org/docs/latest/opensearch/snapshots/snapshot-restore#amazon-s3).
```
Comment (gkamat) [inline nyc_taxis/TEST_PROCEDURES.md:9]: that is stored in ..
storing the snapshot and credentials for an AWS account that has permission..

File: nyc_taxis/TEST_PROCEDURES.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+Searchable Snapshots feature is supported by OpenSearch since version 2.4.0, see the [OpenSearch docs](https://opensearch.org/docs/2.4/opensearch/snapshots/searchable_snapshot) to learn more.
```
Comment (gkamat) [inline nyc_taxis/TEST_PROCEDURES.md:11]: The Searchable Snapshots feature ...

File: nyc_taxis/TEST_PROCEDURES.md [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+
+### Parameters
+
+#### The test procedure allows to overwrite the following parameters using `--workload-params`:
```
Comment (gkamat) [inline nyc_taxis/TEST_PROCEDURES.md:15]: allows the following parameters to be specified using --workload-params

File: nyc_taxis/TEST_PROCEDURES.md [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+
+Additional parameters other than those mentioned in [README](README.md) of NYC taxis workload:
```
Comment (gkamat) [inline nyc_taxis/TEST_PROCEDURES.md:17]: allows the following parameters to be specified using --workload-params in addition to those mentioned in ...

File: nyc_taxis/README.md (hunk 5 above)
Comment (tlfeng) [reply nyc_taxis/README.md:78]: Hi @gkamat, thanks for your careful review! 👍 I will make the changes.

File: nyc_taxis/TEST_PROCEDURES.md [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+* `snapshot_repository_name` (default: "test-repository"): Name of the snapshot repository.
+* `snapshot_name` (default: "test-snapshot"): Name of the snapshot. 
+  It's recommended to assign a different value for different test executions, because there is no operation defined by opensearch-benchmark to delete a snapshot, and new snapshot won't be created if a snapshot with the same name exists in the repository.
```
Comment (gkamat) [inline nyc_taxis/TEST_PROCEDURES.md:20]: It is recommended to ...
and a new snapshot will not be created if a previous snapshot with ...

Also, have you considered adding a custom operation to delete the snapshot [on the lines of this example?](https://github.com/opensearch-project/opensearch-benchmark-workloads/blob/main/http_logs/workload.py#L8-L16)

File: nyc_taxis/TEST_PROCEDURES.md [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+* `s3_bucket_name`: Name of the Amazon S3 bucket that stores the snapshot. The S3 bucket needs to be prepared manually.
+* `s3_bucket_region`: The AWS Region where the Amazon S3 bucket exists. For example, "us-east-1".
+
+#### The test procedure requires to provide parameters for `repository-s3` plugin using `--plugin-params`:
```
Comment (gkamat) [inline nyc_taxis/TEST_PROCEDURES.md:24]: The test procedure requires parameters to be provided for the `repository-s3 plugin`, using ...

File: nyc_taxis/TEST_PROCEDURES.md [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+See the [Benchmark docs](https://github.com/opensearch-project/opensearch-benchmark/blob/main/osbenchmark/resources/provision_configs/main/plugins/v1/repository_s3/README.md
+) for detail.
```
Comment (gkamat) [inline nyc_taxis/TEST_PROCEDURES.md:26]: for details.

File: nyc_taxis/TEST_PROCEDURES.md [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+```
+{
+  "s3_client_name": "default",
+  "s3_access_key": "your AWS access key",
+  "s3_secret_key": "your AWS secret key"
+}
+ ```
+Save it as `params.json` and provide it to Benchmark with `--opensearch-plugins="repository-s3" --plugin-params="/path/to/params.json"`.
+
+#### The test procedure requires to provide parameters for the "provision_config_instance" using `--provision-config-instance-params`:
```
Comment (gkamat) [inline nyc_taxis/TEST_PROCEDURES.md:38]: requires parameters to be provided for the ...

File: nyc_taxis/TEST_PROCEDURES.md [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+
+A "provision_config_instance" is a specific configuration of OpenSearch. The parameter is used for setting feature flag in jvm options to enable experimental feature (for OpenSearch 2.4), and set the node role `search`.
+
+Note that built-in instances can be seen from the [Benchmark repository](https://github.com/opensearch-project/opensearch-benchmark/tree/main/osbenchmark/resources/provision_configs/main/provision_config_instances/v1), and the parameter usage can be seen [here](https://github.com/opensearch-project/opensearch-benchmark/blob/main/osbenchmark/resources/provision_configs/main/provision_config_instances/v1/vanilla/README.md) in the same repository.
```
Comment (gkamat) [inline nyc_taxis/TEST_PROCEDURES.md:42]: Note that use of built-in instances can be seen at ...

File: nyc_taxis/operations/default.json [hunk 17]
```diff
@@ -25,6 +25,47 @@
...
+      "body": {
+        "indices": ["nyc_taxis"]
+      }
+    },
+    {
+      "name": "wait-for-snapshot-create",
+      "operation-type": "wait-for-snapshot-create",
+      "repository": "{{ snapshot_repository_name | default('test-repository') }}",
+      "snapshot": "{{ snapshot_name | default('test-snapshot') }}"
+    },
```
Comment (gkamat) [inline nyc_taxis/operations/default.json:54]: Suggest renaming this to "wait-for-snapshot-creation" to match the "wait-for-completion" operation below.

File: nyc_taxis/test_procedures/default.json [hunk 18]
```diff
@@ -293,5 +293,150 @@
...
+          ,"target-throughput": 1.5
+          {%- elif target_throughput is string and target_throughput.lower() == 'none' %}
+          {%- else %}
+          ,"target-throughput": {{ target_throughput | tojson }}
+          {%- endif %}
+          {%-if search_clients is defined and search_clients %}
+          ,"clients": {{ search_clients | tojson}}
+          {%- endif %}
+        }
+      ]
```
Comment (gkamat) [inline nyc_taxis/test_procedures/default.json:441]: Can you consider putting the common queries into a separate file and having both test-procedures `append-no-conflicts` and `searchable-snapshot` share them to avoid duplication?  Also, please rename `searchable_snapshot` to `searchable-snapshot` for consistency.

File: nyc_taxis/TEST_PROCEDURES.md [hunk 19]
```diff
@@ -0,0 +1,55 @@
...
+  "s3_client_name": "default",
+  "s3_access_key": "your AWS access key",
+  "s3_secret_key": "your AWS secret key"
+}
+ ```
+Save it as `params.json` and provide it to Benchmark with `--opensearch-plugins="repository-s3" --plugin-params="/path/to/params.json"`.
+
+#### The test procedure requires to provide parameters for the "provision_config_instance" using `--provision-config-instance-params`:
+
+A "provision_config_instance" is a specific configuration of OpenSearch. The parameter is used for setting feature flag in jvm options to enable experimental feature (for OpenSearch 2.4), and set the node role `search`.
```
Comment (gkamat) [inline nyc_taxis/TEST_PROCEDURES.md:40]: for setting a feature flag in the JVM options to enable an experimental feature (for OpenSearch 2.4), and to set the node role  to `search`.

**Review by gkamat — CHANGES_REQUESTED** [review]
Added  some suggestions for text-edits and a couple of other recommended changes.

File: nyc_taxis/test_procedures/default.json (hunk 18 above)
Comment (tlfeng) [reply nyc_taxis/test_procedures/default.json:441]: I noticed the obvious code duplication of the `target_throughput` section. 👍 I've changed them in a previous commit https://github.com/opensearch-project/opensearch-benchmark-workloads/pull/58/commits/2a7138d050b645a0204334ad9f0a252d2b494889, by using [macros](https://jinja.palletsprojects.com/en/3.1.x/templates/#macros) syntax, but I changed those back eventually and decided to create a separate PR to make the change throughout the whole code repository, because there are lots of occurrences of the duplication in other files. The duplication of `target_throughput` are introduced by this PR https://github.com/opensearch-project/opensearch-benchmark-workloads/pull/47, it's kind of a bad change.
Do you think it's reasonable that I keep the duplication as it is in this PR, and create another PR to resolve the duplication of `target_throughput` throughout the whole repository.

File: nyc_taxis/README.md (hunk 5 above)
Comment (tlfeng) [reply nyc_taxis/README.md]: Updated the wording according to your suggestions in a new commit https://github.com/opensearch-project/opensearch-benchmark-workloads/pull/58/commits/c8cfc9d7953875a8e2b539191443b477d247b5ea. Thank you!

File: nyc_taxis/TEST_PROCEDURES.md (hunk 12 above)
Comment (tlfeng) [reply nyc_taxis/TEST_PROCEDURES.md]: @gkamat Thanks for pointing the code to me about adding a custom operation, I didn't know that, and will take a look at.

File: nyc_taxis/TEST_PROCEDURES.md (hunk 19 above)
Comment (tlfeng) [reply nyc_taxis/TEST_PROCEDURES.md]: Updated the wording according to your suggestions in a new commit https://github.com/opensearch-project/opensearch-benchmark-workloads/pull/58/commits/c8cfc9d7953875a8e2b539191443b477d247b5ea. Thank you!

File: nyc_taxis/TEST_PROCEDURES.md (hunk 12 above)
Comment (tlfeng) [reply nyc_taxis/TEST_PROCEDURES.md]: Added the custom operation in a new commit https://github.com/opensearch-project/opensearch-benchmark-workloads/pull/58/commits/f62e62883d8d365fa621be6d87b835eaf53a06da. 😄

**Comment by gkamat** [comment]
Will look at the updated changes shortly.

File: nyc_taxis/TEST_PROCEDURES.md (hunk 12 above)
Comment (gkamat) [reply nyc_taxis/TEST_PROCEDURES.md]: Looks good.   Please submit a PR for this when ready.

File: nyc_taxis/test_procedures/default.json (hunk 18 above)
Comment (gkamat) [reply nyc_taxis/test_procedures/default.json:444]: Yes, I'd leave the duplication as is for now, to keep this PR focused on Searchable Snapshots.  Thank you for creating the other issue, which we can address separately.

**Review by gkamat — APPROVED** [review]
Thank you for addressing the comments, @tlfeng.  We can clean up the duplication as a separate update.

File: nyc_taxis/test_procedures/default.json (hunk 18 above)
Comment (tlfeng) [reply nyc_taxis/test_procedures/default.json:444]: @gkamat Thank you for your time in the review! I can handle the code refactor in next week.

File: nyc_taxis/TEST_PROCEDURES.md (hunk 12 above)
Comment (tlfeng) [reply nyc_taxis/TEST_PROCEDURES.md]: Thank you! But no more PR is needed, the code change of the commit https://github.com/opensearch-project/opensearch-benchmark-workloads/pull/58/commits/f62e62883d8d365fa621be6d87b835eaf53a06da has already been in this PR.

**Review by IanHoang — APPROVED** [review]
Sorry it took a while for me to take a look into this but I read through the changes and all comments! Feel free to ping me directly to speed up the review process.
